doc/report/TUDreport.tex

```diff
@@ -0,0 +1,538 @@
+
+  \subsection{Web Application}
+  \label{sec:tech:implementation:bootstrapping:web}
+
+    As previously mentioned~(\ref{sec:tech:implementation:dirs}), the \emph{/client} directory is exposed through the HTTP web server. In particular, the client is eventually sending an HTTP request to the root of this directory. Because no particular file is requested by the client, the server will respond with the contents of the \emph{index.html} file.
+
+    The \emph{index.html} file consists of a simple HTML5 skeleton which basically includes lots of CSS and JavaScript files. One of the libraries that gets included by the file is \emph{AngularJS}. Once the browser is done with parsing the page and loading all embedded JavaScript files, the \emph{AngularJS} framework is scanning for an \texttt{ng-app} attribute inside the DOM. When found, AngularJS automatically starts its bootstrap routine and eventually executes the application, which is happen to be defined inside \emph{js/app.js}. Section~\ref{sec:tech:implementation:angularjs} covers the contents of this file.
+
+  \section{StrongLoop}
+  \label{sec:tech:implementation:strongloop}
+
+    As explained in section~\ref{sec:tech:architecture:3rd-party}, \emph{StrongLoop} is the tool that the application uses to execute CRUD operations on the database. Besides utilizing \emph{StrongLoop} as an ORM, the application makes also use of its other features such as authentication- \& session management, access control and file upload support. The application is mainly managed by the command line tool, \texttt{slc}, that comes together with \emph{StrongLoop}. It allows to generate and extend models, define access control rules and more things. For an exhaustive description of the tool please refer to the official documentation.\footnote{\url{https://docs.strongloop.com/display/public/LB/Command-line+reference}}
+
+  \subsection{Creating and Updating Models}
+  \label{sec:tech:implementation:strongloop:models}
+
+    \emph{StrongLoop} does represent database tables in terms of models. A model does represent a collection of data with a given schema, e.g., a collection of users that have properties like a name, an email, a password, etc. Besides the sheer representation of data, the developer can define access rules in the model that define which data can be accessed by whom. In particular, this feature allows to restrict data access {} the data (see~\ref{sec:tech:implementation:acl:strongloop-acl}). That said, a model does also provide the methods required to retrieve, store, update and delete items of a collections.
```

Comment:
- ... restrict data access to users.. (the data?)
- ... items of a collection. (s muss weg)